salt_sproxy/_modules/netbox.py

```diff
@@ -125,6 +123,4 @@ def get(app, endpoint, id=None, **kwargs):
         return dict(getattr(getattr(nb, app), endpoint).get(id))
     else:
         clean_kwargs = __utils__['args.clean_kwargs'](**kwargs)
-        return dict(
-            getattr(getattr(nb, app), endpoint).get(**clean_kwargs)
-        )
+        return dict(getattr(getattr(nb, app), endpoint).get(**clean_kwargs))
```

Comment:
This is pretty scary as it is very fragile and might be a great source of strange bugs. I would rather break that to defaults and fail, if a method was not found, something like:

``python
ret = None
app = getattr(nb, app, None)
if app is not None:
    endpoint = getattr(app, endpoint, None)
    if endpoint is not None:
        ret = dict(endpoint.get(**clean_kwargs))
    else:
        log.error("Suspicious pointer corrupted virtual machine")
else:
    log.error("The recent proliferation of Nuclear Testing")
return ret

Reply:
I hear ya, I agree. However, this module is simply ported from https://github.com/saltstack/salt/blob/develop/salt/modules/netbox.py with some minor corrections (that are in some PRs that didn't get too much love). Ideally, this module shouldn't be here at all, but it's kinda required and I thought I'd port it with a minimum of changes from the "official" one.